dash-renderer/src/observers/executedCallbacks.ts

```diff
@@ -0,0 +1,219 @@
+
+import {
+    aggregateCallbacks,
+    addRequestedCallbacks,
+    removeExecutedCallbacks,
+    addCompletedCallbacks,
+    addStoredCallbacks
+} from '../actions/callbacks';
+
+import { parseIfWildcard, getCallbacksByInput } from '../actions/dependencies';
+
+import {
+    combineIdAndProp,
+    getLayoutCallbacks,
+    includeObservers
+} from '../actions/dependencies_ts';
+
+import {
+    ICallback,
+    IStoredCallback
+} from '../types/callbacks';
+
+import { updateProps, setPaths, handleAsyncError } from '../actions';
+import { getPath, computePaths } from '../actions/paths';
+
+import {
+    applyPersistence,
+    prunePersistence
+} from '../persistence';
+import { IStoreObserverDefinition } from '../StoreObserver';
+
+const observer: IStoreObserverDefinition<IStoreState> = {
+    observer: ({
+        dispatch,
+        getState
+    }) => {
+        const {
+            callbacks: {
+                executed
+            }
+        } = getState();
+
+        function applyProps(id: any, updatedProps: any) {
+            const { layout, paths } = getState();
+            const itempath = getPath(paths, id);
+            if (!itempath) {
+                return false;
+            }
+
+            // This is a callback-generated update.
+            // Check if this invalidates existing persisted prop values,
+            // or if persistence changed, whether this updates other props.
+            updatedProps = prunePersistence(
+                path(itempath, layout),
+                updatedProps,
+                dispatch
+            );
+
+            // In case the update contains whole components, see if any of
+            // those components have props to update to persist user edits.
+            const { props } = applyPersistence({ props: updatedProps }, dispatch);
+
+            dispatch(
+                updateProps({
+                    itempath,
+                    props,
+                    source: 'response',
+                })
+            );
+
+            return props;
+        }
+
+        let requestedCallbacks: ICallback[] = [];
+        let storedCallbacks: IStoredCallback[] = [];
+
+        forEach(cb => {
+            const {
+                callback: {
+                    clientside_function,
+                    output
+                },
+                executionResult
+            } = cb;
+
+            if (isNil(executionResult)) {
+                return;
+            }
+
+            const { data, error, payload } = executionResult;
+
+            if (data !== undefined) {
+                forEach(([id, props]: [any, { [key: string]: any }]) => {
+                    const parsedId = parseIfWildcard(id);
+                    const { graphs, layout: oldLayout, paths: oldPaths } = getState();
+
+                    // Components will trigger callbacks on their own as required (eg. derived)
+                    const appliedProps = applyProps(parsedId, props);
```

Comment:
Implementation here is adapted from https://github.com/plotly/dash/pull/1254/files#diff-15692ba768eeab56e95aa409ab7c19b7L308